sdk/core/azure-core-http-netty/src/main/java/com/azure/core/http/netty/implementation/NettyAsyncHttpResponse.java

```diff
@@ -58,6 +64,34 @@ public Mono<InputStream> getBodyAsInputStream() {
         return bodyIntern().aggregate().asInputStream();
     }
 
+    @Override
+    public Mono<Void> transferBodyToAsync(AsynchronousByteChannel channel) {
+        return bodyIntern().retain()
+            .flatMapSequential(nettyBuffer ->
+                FluxUtil.writeToAsynchronousByteChannel(Mono.just(nettyBuffer.nioBuffer()).flux(), channel)
+                    .doFinally(ignored -> nettyBuffer.release()), 1, 1)
+            .then();
+    }
+
+    @Override
+    public void transferBodyTo(WritableByteChannel channel) {
+        bodyIntern().retain()
```

Comment:
I don't think retain is needed here are the `ByteBuf` usage completes after the full write has happened (the while loop in map). Retaining should only be necessary in async as the `ByteBuf` goes on to a write stack that could complete after `onNext` is triggered and frees the `ByteBuf`.